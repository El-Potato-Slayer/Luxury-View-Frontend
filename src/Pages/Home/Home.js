import 'pure-react-carousel/dist/react-carousel.es.css';
import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { getRandomElements, sortMansionByNewest } from '../../Helpers/index';
import {
  displayAsSeenIn, displayHeader, displayNewestProperties, displaySoonToCome,
} from './helper';

function Home() {
  const [randomMansions, setRandomMansions] = useState([]);
  const [newestMansions, setNewestMansions] = useState([]);
  const { properties: mansions, loading } = useSelector((state) => state.propertiesReducer);

  useEffect(() => {
    setNewestMansions(sortMansionByNewest(mansions).slice(0, 4));
    setRandomMansions(getRandomElements(mansions, 3));
  }, [mansions]);

  return (
    <div data-testid="homepage">
      {displayHeader(loading, randomMansions)}
      {displayNewestProperties(loading, newestMansions)}
      {displaySoonToCome()}
      {displayAsSeenIn()}
    </div>
  );
}

export default Home;
