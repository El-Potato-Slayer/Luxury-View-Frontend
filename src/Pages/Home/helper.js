import {
  CarouselProvider, DotGroup, Slide, Slider,
} from 'pure-react-carousel';
import { Link } from 'react-router-dom';
import { priceOutput } from '../../Helpers';
import SkeletonHeader from '../../Skeletons/SkeletonHome/SkeletonHeader';
import SkeletonRectangle from '../../Skeletons/SkeletonHome/SkeletonRectangle';

export function displayHeader(isLoading, properties) {
  return (
    <div className="homepage-header">
      {isLoading && <SkeletonHeader />}
      {!isLoading
        && (
        <CarouselProvider
          naturalSlideWidth={134}
          naturalSlideHeight={60}
          totalSlides={properties.length}
          className="carousel"
          isPlaying
        >
          <h2 className="carousel-title">
            The World&apos;s
            <br />
            Luxury Marketplace
          </h2>
          <Slider className="slider">
            {properties.map((mansion, index) => (
              <Slide className="slide" key={mansion.id} index={index} style={{ background: `url(${mansion.picture}) center / cover` }} />
            ))}
          </Slider>
          <DotGroup className="dots" />
        </CarouselProvider>
        )}
    </div>
  );
}

export function displayNewestProperties(isLoading, properties) {
  return (
    <section className="home-section" data-testid="homepage-newest">
      <h4 className="home-section-title">Newest</h4>
      <div className="home-newest">
        {isLoading && (
        <>
          <SkeletonRectangle />
          <SkeletonRectangle />
          <SkeletonRectangle />
          <SkeletonRectangle />
        </>
        )}
        {properties.map((mansion) => (
          <Link key={mansion.id} to={`mansions/${mansion.id}`} style={{ background: `url(${mansion.picture}) center / cover` }} className="home-newest-item">
            <p className="home-mansion-price">
              $
              {priceOutput(mansion.price)}
            </p>
            <p className="home-mansion-name">{mansion.name}</p>
          </Link>
        ))}
      </div>
    </section>
  );
}

export function displaySoonToCome() {
  return (
    <section className="home-section" data-testid="homepage-soon">
      <h4 className="home-section-title">Soon to come</h4>
      <div className="home-newest">
        <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/cars%2Fmercedes_benz_biome_concept_4.jpg?alt=media&token=f5a523f3-49b9-4b30-83f0-b6375c938084) center/ cover' }}>
          <p className="home-mansion-price">Cars</p>
        </div>
        <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/yacht%2Fyacht.jpg?alt=media&token=19b88a2e-92a3-4f1a-9882-c5d7a8074721) center/ cover' }}>
          <p className="home-mansion-price">Yachts</p>
        </div>
        <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/jets%2Fjet.jpg?alt=media&token=1ca94a0c-96af-4a82-9aac-187891d92b48) center/ cover' }}>
          <p className="home-mansion-price">Jets</p>
        </div>
        <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/helicopters%2FHill-Helicopters-HX50-1-1536x864.jpg?alt=media&token=a9412f86-5489-4ebf-a469-e34a55820598) center/ cover' }}>
          <p className="home-mansion-price">Helicopters</p>
        </div>
      </div>
    </section>
  );
}

export function displayAsSeenIn() {
  return (
    <section className="home-section home-seen" data-testid="homepage-seen">
      <h4>As seen in</h4>
      <div className="home-seen-grid">
        <div>
          <img src="/assets/huffpost.svg" alt="huffpost-logo" />
        </div>
        <div>
          <img src="/assets/business-insider.svg" alt="business-insider-logo" />
        </div>
        <div>
          <img src="/assets/bloomberg.svg" alt="bloomberg-logo" />
        </div>
        <div>
          <img src="/assets/cnbc.svg" alt="cnbc-logo" />
        </div>
        <div>
          <img src="/assets/bbc.svg" alt="bbc-logo" />
        </div>
        <div>
          <img src="/assets/financial-times.svg" alt="financial-times-logo" />
        </div>
      </div>
    </section>
  );
}
