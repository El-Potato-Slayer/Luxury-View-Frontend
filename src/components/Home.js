import PropTypes from 'prop-types';
import {
  CarouselProvider, Slider, Slide, DotGroup,
} from 'pure-react-carousel';
import 'pure-react-carousel/dist/react-carousel.es.css';
import { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getRandomElements, priceOutput, sortMansionByNewest } from '../helpers';

function Home({ mansions }) {
  const [randomMansions, setRandomMansions] = useState([]);
  const [newestMansions, setNewestMansions] = useState([]);

  useEffect(() => {
    setNewestMansions(sortMansionByNewest(mansions).slice(0, 4));
    setRandomMansions(getRandomElements(mansions, 3));
  }, [mansions]);

  return (
    <div data-testid="homepage">
      <div className="homepage-header">
        <CarouselProvider
          naturalSlideWidth={134}
          naturalSlideHeight={60}
          totalSlides={randomMansions.length}
          className="carousel"
          isPlaying
        >
          <h2 className="carousel-title">
            The World&apos;s
            <br />
            Luxury Marketplace
          </h2>
          <Slider className="slider">
            {randomMansions.map((mansion, index) => (
              <Slide className="slide" key={mansion.id} index={index} style={{ background: `url(${mansion.picture}) center / cover` }} />
            ))}
          </Slider>
          <DotGroup className="dots" />
        </CarouselProvider>

      </div>

      <section className="home-section" data-testid="homepage-newest">
        <h4 className="home-section-title">Newest</h4>
        <div className="home-newest">
          {newestMansions.map((mansion) => (
            <Link key={mansion.id} to={`mansions/${mansion.id}`} style={{ background: `url(${mansion.picture}) center / cover` }} className="home-newest-item">
              <p className="home-mansion-price">
                $
                {priceOutput(mansion.price)}
              </p>
              <p className="home-mansion-name">{mansion.name}</p>
            </Link>
          ))}
        </div>
      </section>

      <section className="home-section" data-testid="homepage-soon">
        <h4 className="home-section-title">Soon to come</h4>
        <div className="home-newest">
          <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/cars%2Fmercedes_benz_biome_concept_4.jpg?alt=media&token=f5a523f3-49b9-4b30-83f0-b6375c938084) center/ cover' }}>
            <p className="home-mansion-price">Cars</p>
          </div>
          <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/yacht%2Fyacht.jpg?alt=media&token=19b88a2e-92a3-4f1a-9882-c5d7a8074721) center/ cover' }}>
            <p className="home-mansion-price">Yachts</p>
          </div>
          <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/jets%2Fjet.jpg?alt=media&token=1ca94a0c-96af-4a82-9aac-187891d92b48) center/ cover' }}>
            <p className="home-mansion-price">Jets</p>
          </div>
          <div className="home-newest-item" style={{ background: 'url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/helicopters%2FHill-Helicopters-HX50-1-1536x864.jpg?alt=media&token=a9412f86-5489-4ebf-a469-e34a55820598) center/ cover' }}>
            <p className="home-mansion-price">Helicopters</p>
          </div>
        </div>
      </section>

      <section className="home-section home-seen" data-testid="homepage-seen">
        <h4>As seen in</h4>
        <div className="home-seen-grid">
          <div>
            <img src="/assets/huffpost.svg" alt="huffpost-logo" />
          </div>
          <div>
            <img src="/assets/business-insider.svg" alt="business-insider-logo" />
          </div>
          <div>
            <img src="/assets/bloomberg.svg" alt="bloomberg-logo" />
          </div>
          <div>
            <img src="/assets/cnbc.svg" alt="cnbc-logo" />
          </div>
          <div>
            <img src="/assets/bbc.svg" alt="bbc-logo" />
          </div>
          <div>
            <img src="/assets/financial-times.svg" alt="financial-times-logo" />
          </div>
        </div>
      </section>
    </div>
  );
}

const mapStateToProps = (state) => ({
  mansions: state.propertiesReducer.properties,
});

Home.propTypes = {
  mansions: PropTypes.instanceOf(Array).isRequired,
};

// export default Home;
export default connect(mapStateToProps)(Home);
