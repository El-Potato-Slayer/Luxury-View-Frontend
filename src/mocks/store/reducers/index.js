import { combineReducers } from 'redux';
import agentsReducer from './agentsReducer';
import appointmentsReducer from './appointmentsReducer';
import authReducer from './authReducer';
import propertiesReducer from './propertiesReducer';
import userReducer from './userReducer';
import filterReducer from './filterReducer';

export default combineReducers({
  agentsReducer, appointmentsReducer, authReducer, propertiesReducer, userReducer, filterReducer,
});
