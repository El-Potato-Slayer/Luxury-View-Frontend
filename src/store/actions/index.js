export const setAgents = (payload) => ({
  type: 'SET_AGENTS',
  payload,
});

export const setAppointments = (payload) => ({
  type: 'SET_APPOINTMENTS',
  payload,
});

export const setProperties = (payload) => ({
  type: 'SET_PROPERTIES',
  payload,
});

export const changeFilter = (name) => ({
  type: 'CHANGE_FILTER',
  name,
});
