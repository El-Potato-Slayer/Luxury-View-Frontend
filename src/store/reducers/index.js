import { combineReducers } from 'redux';
import agentsReducer from './agentsReducer';
import appointmentsReducer from './appointmentsReducer';
import propertiesReducer from './propertiesReducer';
import userReducer from './userReducer';
import filterReducer from './filterReducer';

export default combineReducers({
  agentsReducer, appointmentsReducer, propertiesReducer, userReducer, filterReducer,
});
