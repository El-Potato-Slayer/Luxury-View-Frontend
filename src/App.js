import './App.css';
import {
  BrowserRouter, Redirect, Route, Switch,
} from 'react-router-dom';
import { connect, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import Navbar from './components/Navbar';
import Home from './components/Home';
import { setAgents, setAuth, setProperties } from './store/actions';
import MansionsList from './containers/MansionsList';
import AgentsList from './containers/AgentsList';
import ProtectedRoute from './components/ProtectedRoute';
import AppointmentsList from './containers/AppointmentsList';
import Login from './components/Login';
import Mansion from './components/Mansion';

function App({ isAuth }) {
  const dispatch = useDispatch();
  let token;
  useEffect(() => {
    axios.get('http://localhost:3000/api/v1/properties')
      .then((resp) => {
        dispatch(setProperties(resp.data));
      });
    axios.get('http://localhost:3000/api/v1/agents')
      .then((resp) => {
        dispatch(setAgents(resp.data));
      });
    token = localStorage.getItem('token');
    if (token) {
      const tokenExpiration = JSON.parse(atob(token.split('.')[1])).exp;
      const dateNow = new Date();
      if (tokenExpiration > dateNow.getTime() / 1000) {
        dispatch(setAuth(true));
      }
    }
  }, []);
  return (
    <div data-testid="app" className="App">
      <BrowserRouter>
        <Navbar />
        <Switch>
          <Route exact path="/">
            <Home />
          </Route>
          <Route exact path="/mansions">
            <MansionsList />
          </Route>
          <Route exact path="/agents">
            <AgentsList />
          </Route>
          <Route path="/login">
            {isAuth ? <Redirect to="/appointments" /> : <Login />}
          </Route>
          <Route path="/mansions/:id">
            <Mansion />
          </Route>
          <ProtectedRoute exact path="/appointments" component={AppointmentsList} />
        </Switch>
      </BrowserRouter>
    </div>
  );
}

const mapStateToProps = (state) => ({
  isAuth: state.authReducer.isAuth,
});

App.propTypes = {
  isAuth: PropTypes.bool.isRequired,
};

export default connect(mapStateToProps)(App);
