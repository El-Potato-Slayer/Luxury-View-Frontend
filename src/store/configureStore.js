import { applyMiddleware, createStore } from 'redux';
import ReduxThunk from 'redux-thunk';
import { fetchData } from './actions';
import { fetchAgentsFailure, fetchAgentsRequest, fetchAgentsSuccess } from './actions/agentActions';
import { fetchAppointmentsFailure, fetchAppointmentsRequest, fetchAppointmentsSuccess } from './actions/appointmentActions';
import { fetchPropertiesFailure, fetchPropertiesRequest, fetchPropertiesSuccess } from './actions/propertyActions';
import rootReducer from './reducers';

const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

// store.subscribe(() => { console.log(store.getState()); });
store.dispatch(fetchData('properties', fetchPropertiesRequest, fetchPropertiesSuccess, fetchPropertiesFailure));
store.dispatch(fetchData('agents', fetchAgentsRequest, fetchAgentsSuccess, fetchAgentsFailure));
store.dispatch(fetchData('appointments', fetchAppointmentsRequest, fetchAppointmentsSuccess, fetchAppointmentsFailure, true));

export default store;
