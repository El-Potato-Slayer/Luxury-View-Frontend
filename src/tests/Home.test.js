import {
  cleanup, render, screen,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import store from '../mocks/store/configureStore';
import App from '../App';

afterEach(cleanup);

test('Renders the Home component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <App>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </App>
    </Provider>, div,
  );
});

test('Checks if title is rendered', () => {
  render(
    <Provider store={store}>
      <App>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </App>
    </Provider>,
  );
  expect(screen.getByTestId('homepage')).toHaveTextContent("The World'sLuxury Marketplace");
});

describe('Newest Section', () => {
  test('Checks if Newest section is rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('Newest');
  });

  test('Checks that mansion names are rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('Suganti Vapi');
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('6 Bed Mansion');
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('Videa Casale');
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('Modern Trophy Estate');
  });

  test('Checks that unknown mansion names are not rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('Ooga');
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('Booga');
  });

  test('Checks that mansion prices are rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('$35,000,000');
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('$100,000,000');
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('$26,000,000');
    expect(screen.getByTestId('homepage-newest')).toHaveTextContent('$2,500,000');
  });

  test('Checks that unknown mansion prices are not rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$42');
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$1.78');
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$547');
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$425');
  });

  test('Checks if mansion prices are not rendered without comma in Newest section', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$35000000');
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$100000000');
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$26000000');
    expect(screen.getByTestId('homepage-newest')).not.toHaveTextContent('$2500000');
  });
});

describe('Soon To Come section', () => {
  test('Checks if Soon To Come section is rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-soon')).toHaveTextContent('Soon');
  });

  test('Checks that card titles are rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-soon')).toHaveTextContent('Cars');
    expect(screen.getByTestId('homepage-soon')).toHaveTextContent('Yachts');
    expect(screen.getByTestId('homepage-soon')).toHaveTextContent('Jets');
    expect(screen.getByTestId('homepage-soon')).toHaveTextContent('Helicopters');
  });

  test('Checks that unknown card titles are not rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-soon')).not.toHaveTextContent('Ooga');
    expect(screen.getByTestId('homepage-soon')).not.toHaveTextContent('Booga');
  });
});

describe('As Seen In section', () => {
  test('Checks if As Seen In section is rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByTestId('homepage-seen')).toHaveTextContent('As seen in');
  });

  test('Checks that images are rendered', () => {
    render(
      <Provider store={store}>
        <App>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </App>
      </Provider>,
    );
    expect(screen.getByAltText('huffpost-logo')).toBeInTheDocument();
    expect(screen.getByAltText('business-insider-logo')).toBeInTheDocument();
    expect(screen.getByAltText('bloomberg-logo')).toBeInTheDocument();
    expect(screen.getByAltText('cnbc-logo')).toBeInTheDocument();
    expect(screen.getByAltText('bbc-logo')).toBeInTheDocument();
    expect(screen.getByAltText('financial-times-logo')).toBeInTheDocument();
  });
});

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <App>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </App>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
