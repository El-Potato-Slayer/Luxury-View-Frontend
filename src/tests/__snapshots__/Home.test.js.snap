// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="app"
>
  <div
    className="navbar"
    data-testid="navbar"
  >
    <div
      aria-hidden="true"
      className="dark-filter hide"
      onClick={[Function]}
      onKeyDown={[Function]}
    />
    <div
      className="burger"
    >
      <button
        className="burger-wrapper"
        id="burger-wrapper"
        onClick={[Function]}
        type="button"
      >
        <div
          className="burger-menu"
          id="burger-menu"
        />
      </button>
    </div>
    <img
      alt=""
      className="nav-logo"
      src="/assets/logo.png"
    />
    <div
      className="nav-links"
    >
      <a
        className="nav-item"
        data-testid="navbar-homepage"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="nav-item"
        data-testid="navbar-mansions"
        href="/mansions"
        onClick={[Function]}
      >
        Mansions
      </a>
      <a
        className="nav-item"
        data-testid="navbar-appointments"
        href="/appointments"
        onClick={[Function]}
      >
        Appointments
      </a>
      <a
        className="nav-item"
        data-testid="navbar-agents"
        href="/agents"
        onClick={[Function]}
      >
        Agents
      </a>
      <a
        className="nav-item"
        data-testid="navbar-login"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        className="nav-item"
        data-testid="navbar-register"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
    <div
      className="nav-social"
    />
  </div>
  <div
    data-testid="homepage"
  >
    <div
      className="homepage-header"
    >
      <div
        className="carousel carousel"
      >
        <h2
          className="carousel-title"
        >
          The World's
          <br />
          Luxury Marketplace
        </h2>
        <div
          aria-live="polite"
          className="horizontalSlider___281Ls carousel__slider carousel__slider--horizontal slider"
          onKeyDown={[Function]}
          role="listbox"
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="carousel__slider-tray-wrapper carousel__slider-tray-wrap--horizontal"
            style={Object {}}
          >
            <ul
              className="sliderTray___-vHFQ sliderAnimation___300FY carousel__slider-tray carousel__slider-tray--horizontal"
              onClickCapture={[Function]}
              onMouseDown={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                  "transform": "translateX(NaN%) translateX(0px)",
                  "width": "0%",
                }
              }
            />
          </div>
        </div>
        <div
          className="carousel__dot-group dots"
        />
      </div>
    </div>
    <section
      className="home-section"
      data-testid="homepage-newest"
    >
      <h4
        className="home-section-title"
      >
        Newest
      </h4>
      <div
        className="home-newest"
      />
    </section>
    <section
      className="home-section"
      data-testid="homepage-soon"
    >
      <h4
        className="home-section-title"
      >
        Soon to come
      </h4>
      <div
        className="home-newest"
      >
        <div
          className="home-newest-item"
          style={
            Object {
              "background": "url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/cars%2Fmercedes_benz_biome_concept_4.jpg?alt=media&token=f5a523f3-49b9-4b30-83f0-b6375c938084) center/ cover",
            }
          }
        >
          <p
            className="home-mansion-price"
          >
            Cars
          </p>
        </div>
        <div
          className="home-newest-item"
          style={
            Object {
              "background": "url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/yacht%2Fyacht.jpg?alt=media&token=19b88a2e-92a3-4f1a-9882-c5d7a8074721) center/ cover",
            }
          }
        >
          <p
            className="home-mansion-price"
          >
            Yachts
          </p>
        </div>
        <div
          className="home-newest-item"
          style={
            Object {
              "background": "url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/jets%2Fjet.jpg?alt=media&token=1ca94a0c-96af-4a82-9aac-187891d92b48) center/ cover",
            }
          }
        >
          <p
            className="home-mansion-price"
          >
            Jets
          </p>
        </div>
        <div
          className="home-newest-item"
          style={
            Object {
              "background": "url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/helicopters%2FHill-Helicopters-HX50-1-1536x864.jpg?alt=media&token=a9412f86-5489-4ebf-a469-e34a55820598) center/ cover",
            }
          }
        >
          <p
            className="home-mansion-price"
          >
            Helicopters
          </p>
        </div>
      </div>
    </section>
    <section
      className="home-section home-seen"
      data-testid="homepage-seen"
    >
      <h4>
        As seen in
      </h4>
      <div
        className="home-seen-grid"
      >
        <div>
          <img
            alt="huffpost-logo"
            src="/assets/huffpost.svg"
          />
        </div>
        <div>
          <img
            alt="business-insider-logo"
            src="/assets/business-insider.svg"
          />
        </div>
        <div>
          <img
            alt="bloomberg-logo"
            src="/assets/bloomberg.svg"
          />
        </div>
        <div>
          <img
            alt="cnbc-logo"
            src="/assets/cnbc.svg"
          />
        </div>
        <div>
          <img
            alt="bbc-logo"
            src="/assets/bbc.svg"
          />
        </div>
        <div>
          <img
            alt="financial-times-logo"
            src="/assets/financial-times.svg"
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;
